OpenRA.Mods.Common/EditorBrushes/EditorTileBrush.cs

```diff
@@ -205,6 +209,10 @@ public void Do()
 
 						mapTiles[c] = new TerrainTile(template, index);
 						mapHeight[c] = (byte)(baseHeight + terrainTemplate[index].Height).Clamp(0, map.Grid.MaximumTerrainHeight);
+						foreach (var preview in editorActorLayer.PreviewsAt(c))
+						{
+							preview.ReplaceInit(new TerrainOrientationInit(map.TerrainOrientation(c)));
+						}
```

Comment:
instead subscribe previews to a CellMap change event. In theory it should be subscribed to both mapHeight and mapRamp cellmaps, but weirdly in this code I can't find it changing the ramp map